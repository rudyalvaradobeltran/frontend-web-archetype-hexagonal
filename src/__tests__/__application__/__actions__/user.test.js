import reducer, { userActions } from '../../../application/actions/user';
import initState from '../../../domain/entities/user';

describe('User Actions', () => {
  it('Reducer: Should the reducer returns the initialState', () => {
    expect(reducer(initState, {})).toMatchSnapshot();
  });

  it('Actions: Should call fetchUsers action', () => {
    expect(reducer(initState, userActions.fetchUsers)).toMatchSnapshot;
  });

  it('Actions: Should call fetchUsersSuccess action', () => {
    expect(
      reducer(
        initState,
        userActions.fetchUsersSuccess([
          {
            address: {},
            company: {},
            email: '',
            id: 0,
            name: '',
            phone: '',
            username: '',
            website: '',
          },
        ])
      )
    ).toMatchSnapshot;
  });

  it('Actions: Should call fetchUsersFailed action', () => {
    expect(reducer(initState, userActions.fetchUsersFailed([]))).toMatchSnapshot;
  });
});
