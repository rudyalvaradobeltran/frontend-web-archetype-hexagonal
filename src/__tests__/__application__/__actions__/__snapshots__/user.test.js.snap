// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Actions Reducer: Should the reducer returns the initialState 1`] = `
Object {
  "errorFetchingUsers": false,
  "errorMsgFetchingUsers": "",
  "users": Array [],
}
`;
